core/embed/rust/src/ui/layout_eckhart/component/hold_to_confirm.rs

```diff
@@ -27,17 +28,50 @@ pub struct HoldToConfirmAnim {
     timer: Stopwatch,
     /// Header overlay text shown during the animation
     header_overlay: Option<TString<'static>>,
+    /// Rollback animation state
+    rollback: RollbackState,
+}
+
+/// State of the rollback animation, when `stop` is called.
+struct RollbackState {
+    /// Timer for the rollback animation
+    timer: Stopwatch,
+    /// Point in time of the growth animation when the rollback was initiated
+    duration: Duration,
 }
 
 impl HoldToConfirmAnim {
+    /// Ratio of total_duration for the bottom part of the border
+    const BOTTOM_DURATION_RATIO: f32 = 0.125;
+    /// Ratio of total_duration for the side parts of the border
+    const SIDES_DURATION_RATIO: f32 = 0.5;
+    /// Ratio of total_duration for the top part of the border
+    const TOP_DURATION_RATIO: f32 = 0.375;
+
+    /// Duration ratio for the rollback animation after `stop` is called
+    const ROLLBACK_DURATION_RATIO: f32 = Self::TOP_DURATION_RATIO;
+
+    /// Duration after which the header overlay is shown after `start` is called
+    const HEADER_OVERLAY_DELAY: Duration = Duration::from_millis(300);
+
     pub fn new() -> Self {
+        // Assert that our ratios sum to 1.0
+        debug_assert!(
+            (Self::BOTTOM_DURATION_RATIO + Self::SIDES_DURATION_RATIO + Self::TOP_DURATION_RATIO
+                - 1.0)
+                < 0.001
+        );
```

Comment:
I have two points here:
- assert catches only cases where the sum is greater than one
- it would be better to have the compile-time error as the sum can be computed in the compile time
My suggestion combines both these points
```suggestion
        const _: () = {
            const SUM: f32 = HoldToConfirmAnim::BOTTOM_DURATION_RATIO
                + HoldToConfirmAnim::SIDES_DURATION_RATIO
                + HoldToConfirmAnim::TOP_DURATION_RATIO;

            assert!(
                SUM >= 0.999 && SUM <= 1.001,
                "Duration ratios do not sum to 1.0"
            );
        };
```

Reply:
I wanted to use `(summation - 1.0).abs() < 0.001` but I found out that `f32::abs` is in `std` and not `core` :( 

I'm considering putting this assert into a unit instead so that we don't over-complicate that :thinking:

Reply:
That is a good idea, let's do it that way

Reply:
b88aec01645f9621f112e4a4205c4a28b1993fa9